### PR TITLE
Jack/episodes+player

### LINKED ISSUE #431: Episode Player
**Description**
Create a player, episode table view

**Designs**
Include any designs here or links to Zeplin designs.

### PR DESCRIPTION
**Pull request by jackthmp**
Created **very basic** SeriesDetailView and collapsable player. Player is copied from old app version, without the episode info (hence all of the constants from previous design)

Closes #431

### CODE REVIEW

File: Recast/EpisodeTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//
+//  Created by Jack Thompson on 9/15/18.
+//
+
+import UIKit
+import SnapKit
+
+class EpisodeTableViewCell: UITableViewCell {
+    
+    /// MARK: Variables
```
Comment (mindylou) [inline Recast/EpisodeTableViewCell.swift]: Can you change these to `// MARK: - ` so they're accessible from the top bar

File: Recast/EpisodeTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        backgroundColor = .white
+        
+        episodeImageView = UIImageView()
+        
+        episodeNameLabel = UILabel()
+        episodeNameLabel.font = UIFont.systemFont(ofSize: 16)
```
Comment (mindylou) [inline Recast/EpisodeTableViewCell.swift]: remove `UIFont.`

File: Recast/EpisodeTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        
+        dateTimeLabel = UILabel()
+        dateTimeLabel.font = UIFont.systemFont(ofSize: 12)
```
Comment (mindylou) [inline Recast/EpisodeTableViewCell.swift]: here too

File: Recast/EpisodeTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        
+        episodeDescriptionView = UILabel()
+        episodeDescriptionView.font = UIFont.systemFont(ofSize: 14)
```
Comment (mindylou) [inline Recast/EpisodeTableViewCell.swift]: same here

File: Recast/EpisodeTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        dateTimeLabel.text = "Jan. 1, 2018 • 23:00"
+        episodeDescriptionView.text = "This is the episode description. This is the episode description. This is the episode description. This is the episode description. This is the episode description. This is the episode description. This is the episode description."
+        
+        layoutSubviews()
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        episodeImageView.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Recast/EpisodeTableViewCell.swift]: remove parentheses around `(make)` for all of the snapkit methods

File: Recast/PlayerHeaderView.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+protocol PlayerHeaderViewDelegate: class {
+    func playerHeaderViewDidTapCollapseButton()
+}
+
+class PlayerHeaderView: UIView {
+    
+    var collapseButton: UIButton!
+    
+    let playerHeaderViewHeight: CGFloat = 55
```
Comment (mindylou) [inline Recast/PlayerHeaderView.swift]: add `// MARK: - ` variables/constants for this file too

File: Recast/PlayerControlsView.swift [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+//
+//  Created by Jack Thompson on 9/18/18.
+//  Copyright © 2018 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+class PlayerControlsView: UIView {
+    
+    let sliderHeight: CGFloat = 1.5
```
Comment (mindylou) [inline Recast/PlayerControlsView.swift]: Can you add a `// MARK: - Variables` and `// MARK: - Constants` here?

File: Recast/PlayerViewController.swift [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+//  Created by Jack Thompson on 9/16/18.
+//  Copyright © 2018 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class PlayerViewController: UIViewController, PlayerHeaderViewDelegate {
+    
+    /// MARK: Variables
```
Comment (mindylou) [inline Recast/PlayerViewController.swift]: change these to `// MARK: -`

File: Recast/PlayerViewController.swift [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        
+        controlsView = PlayerControlsView(frame: .zero)
+        view.addSubview(controlsView)
+        
+        
+        layoutSubviews()
+    }
+    
+    func layoutSubviews() {
+        playerHeaderView.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Recast/PlayerViewController.swift]: remove parentheses around `(make)` here too

File: Recast/PlayerViewController.swift [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    @objc func collapse() {
+        self.dismiss(animated: true, completion: nil)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    /// MARK: PlayerHeaderViewDelegate
```
Comment (mindylou) [inline Recast/PlayerViewController.swift]: make delegates extensions of the view controller and use `// MARK: -`

File: Recast/SeriesHeaderView.swift [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+//  Created by Jack Thompson on 9/16/18.
+//  Copyright © 2018 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class SeriesHeaderView: UIView {
+    
+    /// MARK: Variables
```
Comment (mindylou) [inline Recast/SeriesHeaderView.swift]: change these here too to `// MARK: -`

File: Recast/SeriesHeaderView.swift [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        
+        addSubview(imageView)
+        addSubview(titleLabel)
+        addSubview(publisherLabel)
+        
+        layoutSubviews()
+    }
+    
+    override func layoutSubviews() {
+        imageView.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Recast/SeriesHeaderView.swift]: remove parentheses

File: Recast/SeriesViewController.swift [hunk 13]
```diff
@@ -1,20 +1,80 @@
...
 //  Copyright © 2018 Cornell AppDev. All rights reserved.
 //
 
 import UIKit
+import SnapKit
 
 class SeriesViewController: UIViewController {
-
+    
+    /// MARK: Variables
```
Comment (mindylou) [inline Recast/SeriesViewController.swift]: same thing here

File: Recast/SeriesViewController.swift [hunk 14]
```diff
@@ -1,20 +1,80 @@
...
+        episodeTableView.tableHeaderView = headerView
+        
+        view.addSubview(episodeTableView)
+        
+        
+        layoutSubviews()
+    }
+    
+    func layoutSubviews() {
+        headerView.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Recast/SeriesViewController.swift]: remove parentheses here

File: Recast/SeriesViewController.swift [hunk 15]
```diff
@@ -1,20 +1,80 @@
...
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
     }
 
 }
 
+///MARK: episodeTableView Delegate & DataSource
```
Comment (mindylou) [inline Recast/SeriesViewController.swift]: Can you separate these out into delegate and data source extensions? Use `// MARK - `

**Review by mindylou — CHANGES_REQUESTED** [review]
Some small (but tedious) style changes otherwise looks good! I'll approve once those are fixed

**Review by mindylou — DISMISSED** [review]

File: Recast/EpisodeTableViewCell.swift [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+class EpisodeTableViewCell: UITableViewCell {
+
+    // MARK: - Variables
+    var episodeImageView: UIImageView!
+    var episodeNameLabel: UILabel!
+    var episodeDescriptionView: UILabel!
+    var dateTimeLabel: UILabel!
+
+    // MARK: - Constants
+    let padding = 5
```
Comment (doctor3w) [inline Recast/EpisodeTableViewCell.swift]: Move these to the layoutSubviews function

File: Recast/PlayerControlsView.swift [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+    var backwardsButton: UIButton!
+    var rightTimeLabel: UILabel!
+    var leftTimeLabel: UILabel!
+    //    var recommendButton: FillNumberButton!
+    //    var moreButton: MoreButton!
+    var speedButton: UIButton!
+    var settingsButton: UIButton!
+
+    // MARK: - Constants
+    let sliderHeight: CGFloat = 1.5
```
Comment (doctor3w) [inline Recast/PlayerControlsView.swift]: move all these to layoutSubviews()

File: Recast/PlayerHeaderView.swift [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    func playerHeaderViewDidTapCollapseButton()
+}
+
+class PlayerHeaderView: UIView {
+
+    // MARK: - Variables
+    var collapseButton: UIButton!
+
+    // MARK: - Constants
+    let playerHeaderViewHeight: CGFloat = 55
```
Comment (doctor3w) [inline Recast/PlayerHeaderView.swift]: again move these

File: Recast/PlayerViewController.swift [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+import SnapKit
+
+class PlayerViewController: UIViewController {
+
+    // MARK: - Variables
+    var controlsView: PlayerControlsView!
+    var playerHeaderView: PlayerHeaderView!
+
+    // MARK: - Constants
+    let topPadding: CGFloat = 100
```
Comment (doctor3w) [inline Recast/PlayerViewController.swift]: here too

File: Recast/SeriesHeaderView.swift [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+
+class SeriesHeaderView: UIView {
+
+    // MARK: - Variables
+    var imageView: UIImageView!
+    var titleLabel: UILabel!
+    var publisherLabel: UILabel!
+
+    // MARK: - Constants
+    let imageHeight = 100
```
Comment (doctor3w) [inline Recast/SeriesHeaderView.swift]: these too

File: Recast/SeriesViewController.swift [hunk 21]
```diff
@@ -1,19 +1,81 @@
...
 
 class SeriesViewController: UIViewController {
 
+    // MARK: - Variables
+    var headerView: SeriesHeaderView!
+    var episodeTableView: UITableView!
+
+    // MARK: - Constants
+    let reuseIdentifer = "episodeCell"
+    let headerHeight: CGFloat = 200
```
Comment (doctor3w) [inline Recast/SeriesViewController.swift]: move this

File: Recast/SeriesViewController.swift [hunk 22]
```diff
@@ -1,19 +1,81 @@
...
+import SnapKit
 
 class SeriesViewController: UIViewController {
 
+    // MARK: - Variables
+    var headerView: SeriesHeaderView!
+    var episodeTableView: UITableView!
+
+    // MARK: - Constants
+    let reuseIdentifer = "episodeCell"
```
Comment (doctor3w) [inline Recast/SeriesViewController.swift]: better name for this variable

**Review by doctor3w — CHANGES_REQUESTED** [review]

**Review by mindylou — APPROVED** [review]

**Review by doctor3w — APPROVED** [review]

**Pull request merged**